pydoctor/model.py

```diff
@@ -501,8 +549,66 @@ def setup(self) -> None:
         super().setup()
         self.rawbases: List[str] = []
         self.subclasses: List[Class] = []
+        self._initialbaseobjects = []
+        self._mro: Optional[List[Union['Class', str]]] = None
+    
+    def _init_mro(self) -> None:
+        """
+        Compute the correct value of the method resolution order returned by L{mro()}.
+        """
+        try:
+            self._mro = compute_mro(self)
+        except ValueError as e:
+            self.report(str(e), 'mro')
+            self._mro = list(self.allbases(True))
 
+    @overload
+    def mro(self, include_external:'Literal[True]', include_self:bool=True) -> List[Union['Class', str]]:...
+    @overload
+    def mro(self, include_external:'Literal[False]'=False, include_self:bool=True) -> List['Class']:...
+    def mro(self, include_external:bool=False, include_self:bool=True) -> List[Union['Class', str]]: # type:ignore[misc]
+        """
+        Get the method resution order of this class. 
+
+        @note: The actual correct value is only set in post-processing, if L{mro()} is called
+            in the AST visitors, it will return the same as C{list(self.allbases(True))}.
+        """
+        if self._mro is None:
+            return list(self.allbases(include_self))
+        
+        _mro: List[Union[str, Class]]
+        if include_external is False:
+            _mro = [o for o in self._mro if not isinstance(o, str)]
+        else:
+            _mro = self._mro
+        if include_self is False:
+            _mro = _mro[1:]
+        return _mro
+
+    @property
+    def bases(self) -> List[str]:
+        """
+        Fully qualified names of the bases of this class.
+        """
+        return [self.parent.expandName(str_base) for str_base in self.rawbases]
```

Comment:
This is not a cheap property to access. I believe we could compute it two times only like we do for the resolved bases. Or maybe we actually don't need this attribute at all maybe.